### ISSUE TITLE
Partial download does not result in non-zero error code

### ISSUE THREAD

**Issue by kballard** [guidomb/carthage_cache#32]
If `carthage_cache` fails to download the full archive, it still tries to unzip it, and it ignores any errors reported by the unzip. The output of the tool includes

```
  error:  invalid compressed data to inflate
file #766:  bad zipfile offset (lseek):  137224192
file #767:  bad zipfile offset (local header sig):  137230240
file #768:  bad zipfile offset (local header sig):  137230299
file #769:  bad zipfile offset (local header sig):  137230622
file #770:  bad zipfile offset (local header sig):  137231506
…
```

so I can see it in the log, but `carthage_cache` itself must still be exiting with code `0` because my Makefile continues building the project.

This is with version 0.4.0, but I don't see anything relevant in the subsequent commit log.

**Comment by guidomb** [comment]
You are correct. `carthage_cache` is not checking if the archive was successfully downloaded. That check should be done [here](https://github.com/guidomb/carthage_cache/blob/master/lib/carthage_cache/archive_installer.rb#L18) and the repository should report the error, check [here](https://github.com/guidomb/carthage_cache/blob/master/lib/carthage_cache/repository.rb#L19).

Feel like sending a PR to fix this?

**Comment by kballard** [comment]
I literally haven't touched Ruby in over 8 years, so I'd rather let someone who's a little less rusty submit the PR.
